lib_test/unit_tests.ml

```diff
@@ -1,71 +1,95 @@
 
 let assert_cdp_prep_equal c1 c2 =
-  assert (compareish c1.cdp_value c2.cdp_value);
-  assert (c1.cdp_unknown_pdps = c2.cdp_unknown_pdps)
+  compare_float __LOC__ c1.cdp_value c2.cdp_value;
+  Alcotest.(check int) __LOC__ c1.cdp_unknown_pdps c2.cdp_unknown_pdps
 
 let assert_fring_equal f1 f2 =
   for i=0 to f1.Fring.size - 1 do
-    assert (compareish2 (Fring.peek f1 i) (Fring.peek f2 i))
+    let peek1 = Fring.peek f1 i in
+    let peek2 = Fring.peek f2 i in
+    (* Relax tolerance, representation gets cut off on serialization*)
+    Alcotest.check (float 0.0001) "FRing value" peek1 peek2;
   done
 
 let assert_rra_equal a1 a2 =
   assert (a1.rra_cf = a2.rra_cf);
-  assert (a1.rra_row_cnt = a2.rra_row_cnt);
-  assert (a1.rra_pdp_cnt = a2.rra_pdp_cnt);
-  assert (compareish a1.rra_xff a2.rra_xff);
+  Alcotest.(check int) __LOC__ a1.rra_row_cnt a2.rra_row_cnt;
+  Alcotest.(check int) __LOC__ a1.rra_pdp_cnt a2.rra_pdp_cnt;
+  compare_float __LOC__ a1.rra_xff a2.rra_xff;
   List.iter2 assert_cdp_prep_equal (Array.to_list a1.rra_cdps) (Array.to_list a2.rra_cdps);
   List.iter2 assert_fring_equal (Array.to_list a1.rra_data) (Array.to_list a2.rra_data)
 
 let assert_rras_equal a1s a2s =
   List.iter2 assert_rra_equal (Array.to_list a1s) (Array.to_list a2s)
 
 let assert_rrds_equal r1 r2 =
-  assert (compareish r1.last_updated r2.last_updated);
-  assert (r1.timestep=r2.timestep);
+  compare_float __LOC__ r1.last_updated r2.last_updated;
+  Alcotest.(check int64) __LOC__ r1.timestep r2.timestep;
   assert_dss_equal r1.rrd_dss r2.rrd_dss;
   assert_rras_equal r1.rrd_rras r2.rrd_rras
 
-let test_marshal rrd =
-  "Save to disk" >:: (fun () ->
-      Rrd_unix.to_file rrd "/tmp/output.xml")
-
-let test_unmarshal rrd =
-  "Restore from disk" >:: (fun () ->
-      Rrd_unix.to_file rrd "/tmp/output.xml";
-      let rrd' = Rrd_unix.of_file "/tmp/output.xml" in
-      assert_rrds_equal rrd rrd')
-
-let create_dummy_data () =
+let in_range min max values =
+  let between value =
+    if not (Utils.isnan value) then begin
+      Alcotest.(check bool) (Printf.sprintf "value (%f) higher than min (%f); " value min) (min <= value) true;
+      Alcotest.(check bool) (Printf.sprintf "value (%f) lower than max (%f); " value max) (max >= value) true
+    end in
+  Alcotest.(check bool) (Printf.sprintf "min (%f) ≤ max (%f); " min max) (min <= max) true;
+  List.iter between values
+
+let fring_to_list fring =
+  Array.to_list @@ Fring.get fring
+
+(* Checks if all the values in the archives are within the limits set by the data sources
+ * Each archive (RRA) has an ring for each datasource (DS) *)
+let test_ranges rrd () =
+  let in_range_fring ds fring =
+    in_range ds.ds_min ds.ds_max (fring_to_list fring) in
+  let in_range_rra dss rra =
+    List.iter2 in_range_fring dss (Array.to_list rra.rra_data) in
+
+  List.iter (in_range_rra @@ Array.to_list rrd.rrd_dss) (Array.to_list rrd.rrd_rras)
+
+let test_marshall rrd () =
+  Rrd_unix.to_file rrd "/tmp/output.xml"
```

Comment:
Maybe it is better to do this in the local directory to avoid conflicts. Using `Filename.temp_file` could be also a good idea,